# Analyse statistique

In [18]:
from dataviz_tips.data import load_tips

In [19]:
tips = load_tips()

In [20]:
tips.describe()

,total_bill,tip,size
count,243.000000,243.000000,243.000000
mean,19.813868,3.002387,2.572016
std,8.910071,1.385002,0.952356
min,3.070000,1.000000,1.000000
25%,13.380000,2.000000,2.000000
50%,17.810000,2.920000,2.000000
75%,24.175000,3.575000,3.000000
max,50.810000,10.000000,6.000000


In [21]:
coefficient_mul_bill = tips["total_bill"].max() / tips["total_bill"].quantile(0.75)
coefficient_mul_bill

np.float64(2.101758014477767)

In [22]:
coefficient_mul_tip = tips["tip"].max() / tips["tip"].quantile(0.75)
coefficient_mul_tip

np.float64(2.797202797202797)

## Constat:

- La moyenne de la facture (19,81) est supérieure à la médiane (17,81) : quelques grosses factures étirent la distribution vers la droite (Asymétrie positive)

    - Le max de 50,81 face à la médiane de 17,81 va dans ce sens.
    - Le max de 50,81 est 2,1 fois superieur au quartile 75% (24,12)

- La moyenne du pourboire (3,00) est supérieure à la médiane (2,92) : quelques pourboires étirent la distribution vers la droite (Asymétrie positive)

    - Le max de 10 face à la médiane de 2,92 va dans ce sens.
    - Le max (10) est 2,8 fois superieur au quartile 75% (3,56)

## La largeur de la marge autorisée et intervalle de seuil `[Q1 − marge ; Q3 + marge]`

### 1. Facture

In [23]:
Q1_BILL = tips["total_bill"].quantile(0.25)
Q3_BILL = tips["total_bill"].quantile(0.75)
IQR_BILL = Q3_BILL - Q1_BILL
MARGIN_BILL = 1.5 * IQR_BILL

LIMIT_LEFT_BILL = Q1_BILL - MARGIN_BILL
LIMIT_RIGHT_BILL = Q3_BILL + MARGIN_BILL

print("Seuil inférieur pour les additions:", LIMIT_LEFT_BILL)
print("Seuil supérieur pour les additions:", LIMIT_RIGHT_BILL)

Seuil inférieur pour les additions: -2.8124999999999964
Seuil supérieur pour les additions: 40.36749999999999


In [24]:
print(
    f"Nombre d'observations avec une facture supérieure à {LIMIT_RIGHT_BILL}:",
    tips[tips["total_bill"] > LIMIT_RIGHT_BILL].shape[0],
)

print(
    f"Le pourcentage d'observations avec une facture supérieure à {LIMIT_RIGHT_BILL}:",
    (tips[tips["total_bill"] > LIMIT_RIGHT_BILL].shape[0] / tips["total_bill"].count()),
)

tips[tips["total_bill"] > LIMIT_RIGHT_BILL]

Nombre d'observations avec une facture supérieure à 40.36749999999999: 9
Le pourcentage d'observations avec une facture supérieure à 40.36749999999999: 0.037037037037037035


,total_bill,tip,sex,smoker,day,time,size
59,48.27,6.73,Male,No,Sat,Dinner,4
102,44.30,2.50,Female,Yes,Sat,Dinner,3
142,41.19,5.00,Male,No,Thur,Lunch,5
156,48.17,5.00,Male,No,Sun,Dinner,6
170,50.81,10.00,Male,Yes,Sat,Dinner,3
182,45.35,3.50,Male,Yes,Sun,Dinner,3
184,40.55,3.00,Male,Yes,Sun,Dinner,2
197,43.11,5.00,Female,Yes,Thur,Lunch,4
212,48.33,9.00,Male,No,Sat,Dinner,4


#### Constat: 

- Intervalle d'addition: [-2.81, 40.36]

- L'addition max(50,81) est au delà de la borne haute: 9 additions la dépassent

- Le pourcentage d'observations avec une addition supérieure à 40.36 est de 3.703%

- Les factures avec des valeurs extrêmes supérieur à 40.36 n'ont pas toutes des pourboires élevés

### 2. Pourboire

In [25]:
Q1_TIP = tips["tip"].quantile(0.25)
Q3_TIP = tips["tip"].quantile(0.75)
IQR_TIP = Q3_TIP - Q1_TIP
MARGIN_TIP = 1.5 * IQR_TIP

LIMIT_LEFT_TIP = Q1_TIP - MARGIN_TIP
LIMIT_RIGHT_TIP = Q3_TIP + MARGIN_TIP

print("Seuil inférieur pour les pourboires:", LIMIT_LEFT_TIP)
print("Seuil supérieur pour les pourboires:", LIMIT_RIGHT_TIP)

Seuil inférieur pour les pourboires: -0.36250000000000027
Seuil supérieur pour les pourboires: 5.9375


In [26]:
print(
    f"Nombre d'observations avec un pourboire supérieur à {LIMIT_RIGHT_TIP}:",
    tips[tips["tip"] > LIMIT_RIGHT_TIP].shape[0],
)

print(
    f"Le pourcentage d'observations avec un pourboire supérieur à {LIMIT_RIGHT_TIP}:",
    (tips[tips["tip"] > LIMIT_RIGHT_TIP].shape[0] / tips["tip"].count()),
)

tips[tips["tip"] > LIMIT_RIGHT_TIP]

Nombre d'observations avec un pourboire supérieur à 5.9375: 8
Le pourcentage d'observations avec un pourboire supérieur à 5.9375: 0.03292181069958848


,total_bill,tip,sex,smoker,day,time,size
23,39.42,7.58,Male,No,Sat,Dinner,4
47,32.40,6.00,Male,No,Sun,Dinner,4
59,48.27,6.73,Male,No,Sat,Dinner,4
141,34.30,6.70,Male,No,Thur,Lunch,6
170,50.81,10.00,Male,Yes,Sat,Dinner,3
183,23.17,6.50,Male,Yes,Sun,Dinner,4
212,48.33,9.00,Male,No,Sat,Dinner,4
214,28.17,6.50,Female,Yes,Sat,Dinner,3


In [27]:
total_bill = tips[tips["tip"] > LIMIT_RIGHT_TIP]["total_bill"]
mask = total_bill > tips["total_bill"].median()
mask.all()

np.True_

#### Constat: 
- Intervalle de pourboire: [-0.34, 5.93]

- Le pourboire max(10) est au delà de la borne haute: 8 pourboires la dépassent

- Le pourcentage d'observations avec un pourboire supérieur à 5.93 est de 3.292%

- Les pourboires avec des valeurs extrêmes supérieur à 5.93 ont toutes des factures supérieur à la médiane

### 3. Corrélation entre la Facture et le Pourboire

#### Critère retenu:

**Critère retenu** : une corrélation est jugée stable si elle varie de moins de 0,1 en valeur absolue. Ce seuil est une convention choisie avant le calcul, sans autorité statistique universelle.
    

**Étiquettes de corrélation**: convention adaptée de Cohen (1988)

|\|r\|	      |Intensité    |
|-------------|-------------|
|< 0,1	      |négligeable  |
|0,1 à < 0,3  |faible       |
|0,3 à < 0,5  |moyenne      |
|≥ 0,5	      |forte        |

In [28]:
tips[(tips["total_bill"] > LIMIT_RIGHT_BILL) & (tips["tip"] > LIMIT_RIGHT_TIP)]

,total_bill,tip,sex,smoker,day,time,size
59,48.27,6.73,Male,No,Sat,Dinner,4
170,50.81,10.00,Male,Yes,Sat,Dinner,3
212,48.33,9.00,Male,No,Sat,Dinner,4


In [29]:
lines = tips[(tips["total_bill"] > LIMIT_RIGHT_BILL) | (tips["tip"] > LIMIT_RIGHT_TIP)]
print(lines.shape[0])

14


In [30]:
bill_tip = tips[
    (tips["total_bill"] <= LIMIT_RIGHT_BILL) & (tips["tip"] <= LIMIT_RIGHT_TIP)
]
print(bill_tip.shape[0])

229


In [31]:
bill_tip.corr(numeric_only=True)

,total_bill,tip,size
total_bill,1.000000,0.605171,0.561721
tip,0.605171,1.000000,0.413063
size,0.561721,0.413063,1.000000


#### Constat 1: 

- Nombre d'observations avec une facture supérieure à 40.36 et un pourboire supérieur à 5.93: 3

- Le montant du pourboire est corrélé avec le montant de la facture à 0.605, ce qui indique une relation forte entre les deux variables.

- Le nombre de client par table est corrélé avec le montant de la facture à 0.561, ce qui indique une relation forte entre les deux variables.

- Le montant du pourboire est corrélé avec le nombre de client par table à 0.413, ce qui indique une relation moyenne entre les deux variables.

In [32]:
tips.corr(numeric_only=True)

,total_bill,tip,size
total_bill,1.000000,0.674998,0.597589
tip,0.674998,1.000000,0.488400
size,0.597589,0.488400,1.000000


#### Constat 2:

- Le montant du pourboire est corrélé avec le montant de la facture à 0.674, ce qui indique une relation positive forte

- Le nombre de personnes par table est corrélé avec le montant de la facture à 0.597, ce qui indique une relation positive forte

- Le montant du pourboire est corrélé avec le nombre de personnes par table à 0.488, ce qui indique une relation positive moyenne

#### Conclusion:

- Les trois corrélations varient de moins de 0,1 et conservent le même classement avec et sans valeurs extrêmes. Les extrêmes ne pilotent donc pas la relation observée.

- La légère baisse après retrait des extrêmes est attendue : restreindre l'étendue des variables réduit mécaniquement la corrélation.

### 4. Nombre de personne par table

In [33]:
tips["size"].value_counts()

size
2    155
3     38
4     37
5      5
1      4
6      4
Name: count, dtype: int64

In [34]:
print("Pourcentage de tables avec 1 personne :", (tips["size"] == 1).mean())
print("Pourcentage de tables avec 5 personnes :", (tips["size"] == 5).mean())
print("Pourcentage de tables avec 6 personnes :", (tips["size"] == 6).mean())

Pourcentage de tables avec 1 personne : 0.01646090534979424
Pourcentage de tables avec 5 personnes : 0.0205761316872428
Pourcentage de tables avec 6 personnes : 0.01646090534979424


#### Constat 3:

- Le nombre de personnes par table varie de 1 à 6

- La majorité des tables ont entre 2 et 4 personnes

- Il y a quelques tables avec 1, 5 ou 6 personnes avec des pourcentages faibles (inférieurs ou égaux à 2%), ce qui peut être considéré comme des valeurs extrêmes

### Décisions

- Factures extrêmes (> 40,36 ; 9 lignes, 3,703 %) : rares mais rien n'indique qu'elles soient erronées ; toutes n'ont pas un gros pourboire, ce qui est cohérent avec une relation positive forte. On les garde.

- Pourboires extrêmes (> 5,93 ; 8 lignes, 3,292 %) : tous associés à des factures supérieures à la médiane, ce qui est cohérent avec la relation facture-pourboire. On les garde.

- Extrêmes sur les deux variables (3 lignes) : même raisonnement. Les corrélations varient de moins de 0,1 avec ou sans elles, leur retrait n'apporte rien. On les garde.

- Tables de 1, 5 et 6 personnes : peu fréquentes mais plausibles. On les garde.